File: 2017/day02/python2/advent02.py
#!/usr/bin/python2


with open('../input.txt', 'r') as f:
    lines = f.read().splitlines()

checksum = 0
for line in lines:
    items = line.split('\t')
    max = items[0]
    min = items[0]
    for item in items:
        if int(item) > int(max):
            max = item
        elif int(item) < int(min):
            min = item
    checksum += int(max) - int(min)
print checksum

checksum = 0
for line in lines:
    items = line.split('\t')
    for i in enumerate(items):
        for j in enumerate(items):
            if i[0] == j[0]:
                continue
            item1 = int(i[1])
            item2 = int(j[1])
            if item1 > item2:
                if not item1 % item2:
                    checksum += item1 % item2
                    break
            if item2 > item1:
                if not item2 % item1:
                    checksum += item2 / item1
                    break
print checksum

File: 2017/day02/javascript/advent02.js
#!/usr/bin/node


var fs = require('fs');
var lines = fs.readFileSync('../input.txt', 'utf8').match(/[^\r\n]+/g);

var checksum = 0;
for(var i = 0; i < lines.length; i++)
{
	var items = lines[i].split('\t');
	var max = items[0];
	var min = items[0];
	for(var j = 1; j < items.length; j++)
	{
		if(parseInt(items[j]) > parseInt(max))
			max = items[j];
		else if(parseInt(items[j]) < parseInt(min))
			min = items[j];
	}
	checksum += parseInt(max) - parseInt(min)
}
console.log(checksum);

checksum = 0;
for(var i = 0; i < lines.length; i++)
{
	var items = lines[i].split('\t');
	for(var j = 0; j < items.length; j++)
	{
		for(var k = 0; k < items.length; k++)
		{
			if(items[j] == items[k])
				continue;
			item1 = parseInt(items[j]);
			item2 = parseInt(items[k]);
			if(item1 > item2)
			{
				if((item1 % item2) == 0)
				{
					checksum += item1 % item2;
					break;
				}
			}
			if(item2 > item1)
			{
				if((item2 % item1) == 0)
				{
					checksum += item2 / item1;
					break;
				}
			}
		}
	}
}
console.log(checksum);
